wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,497 @@
+    setIsSnackbarOpen(false);
+  };
+
+  const [snackbarMessage, setSnackbarMessage] = useState(<></>);
+
+  const showSnackbar = (
+    isSuccess: boolean,
+    explorerPath: string,
+    transactionHash: string,
+  ) => {
+    setSnackbarMessage(
+      <>
+        Transaction {isSuccess ? 'succeeded' : 'failed'}:{' '}
+        <Link
+          color="rgb(0, 176, 255)"
+          href={`https://bigdipper.live/${explorerPath}/transactions/${transactionHash}`}
+          target={transactionHash}
+        >
+          ...{transactionHash.slice(transactionHash.length - 12)}
+        </Link>
+      </>,
+    );
+    setIsSnackbarOpen(true);
+  };
+
+  const {
+    connectWithKeplr,
+    setRemoteChainAddress,
+    isRemoteChainAddressInvalid,
+    remoteChainBalance,
+    remoteChainAddress,
+    remoteChainSigner,
+  } = useRemoteChainAccount(purse?.brandPetname);
+
+  const handleAmountChange = e => {
+    setAmount(e.target.value);
+  };
+
+  const isAmountInvalid = useMemo(() => {
+    switch (direction) {
+      case IbcDirection.Withdrawal:
+        try {
+          const val = parseAsValue(
+            amount,
+            purse?.displayInfo?.assetKind,
+            purse?.displayInfo?.decimalPlaces,
+          );
+          return val === 0n || val > (purseBalance ?? 0n);
+        } catch {
+          return true;
+        }
+      case IbcDirection.Deposit:
+        try {
+          const val = parseAsValue(
+            amount,
+            purse?.displayInfo?.assetKind,
+            purse?.displayInfo?.decimalPlaces,
+          );
+          return val === 0n || val > (remoteChainBalance ?? 0n);
+        } catch {
+          return true;
+        }
+    }
+  }, [amount, purse, purseBalance]);
+
+  const close = () => {
+    setInProgress(false);
+    setError('');
+    setAmount('');
+    setRemoteChainAddress('');
+    handleClose();
+  };
+
+  const send = async () => {
+    setError('');
+
+    let val: string;
+    try {
+      assert(ibcAsset);
+      val = String(
+        parseAsValue(
+          amount,
+          purse?.displayInfo?.assetKind,
+          purse?.displayInfo?.decimalPlaces,
+        ),
+      );
+    } catch (e) {
+      setError(String(e));
+      return;
+    }
+
+    setInProgress(true);
+    try {
+      if (direction === IbcDirection.Deposit) {
+        assert(remoteChainSigner && keplrConnection);
+        const {
+          chainInfo: { rpc, gas, explorerPath },
+          deposit,
+        } = ibcAsset;
+
+        const res = await sendIbcTokens(
+          deposit,
+          rpc,
+          remoteChainSigner,
+          val,
+          remoteChainAddress,
+          keplrConnection.address,
+          gas,
+        );
+
+        close();
+        try {
+          assertIsDeliverTxSuccess(res);
+          showSnackbar(true, explorerPath, res.transactionHash);
+        } catch {
+          showSnackbar(false, explorerPath, res.transactionHash);
+        }
+      } else {
+        // Is withdrawal.
```

Comment:
shouldn't need a comment. consider `else if (direction === IbcDirection.Withdrawal)` and an `else` for unknown values. alternately `switch(direction)` and call helper functions

Reply:
Added an `else if`